src/Reflection/ReflectionParameter.php

```diff
@@ -497,8 +496,26 @@ private function getClassName(): ?string
         }
 
         $type = $this->getType();
-        assert($type instanceof ReflectionType);
-        $typeHint = (string) $type;
+        assert($type instanceof ReflectionNamedType || $type instanceof ReflectionUnionType);
+
+        if ($type instanceof ReflectionUnionType) {
+            foreach ($type->getTypes() as $innerType) {
+                assert($innerType instanceof ReflectionNamedType);
```

Comment:
This assertion should IMO be removed, and the type refined on `ReflectionUnionType` instead. Coincidentally, this will expose BC breaks in future PHP versions.